tests/test_edges/test_edge_population.py

```diff
@@ -624,8 +624,8 @@ def test_spatial_synapse_index(self):
 
     @mock.patch.dict(sys.modules, {"spatial_index": mock.Mock()})
     def test_spatial_synapse_index_call(self):
-        self.test_obj.spatial_synapse_index
-        sys.modules["spatial_index"].open_index.assert_called_once_with("path/to/edge/dir")
+        with pytest.raises(KeyError):
```

Comment:
In general we can be more specific with the error adding `match=` to `pytest.raises`,
but in this case I think that it would be more appropriate to raise a BluepySnapError with a more descriptive message, if the property is accessed and it cannot return the index because it's missing.